sos/collector/__init__.py

```diff
@@ -463,6 +465,15 @@ def add_parser_options(cls, parser):
                                  choices=['auto', 'https', 'ftp', 'sftp',
                                           's3'],
                                  help="Manually specify the upload protocol")
+        collect_grp.add_argument('--inherit-config-file', type=str,
+                                 default=None,
+                                 help='Path to a local config file to'
+                                 ' copy to the remote node and use with'
+                                 ' sos report')
```

Comment:
My initial thought was that this would be a toggle, whereby if `True` we implicitly take the config-file used locally, so `sos.conf` by default or whatever gets set.

Is there a use case you have where a user would have config_file_A locally that is used for the `collect` and local `report` collections, but would have config_file_B _also_ local to the system that is running collect but should be used only for remote node collections?

Reply:
I suspect any difference is mostly my misinterpretation of the discussion, not intentional changes to your thought on implementation.

> Is there a use case you have where [...]

Possibly, but not one of my requirements. I plan to use the same config for the collector and the local/remote nodes.

---

I'll attempt to layout the resulting behavior from this change.

### Overview

Three arguments

1. --config-file: str
2. --node-config-file: str
3. --inherit-config-file: bool

### Logic

* `--inherit-config-file` is `False`
  * `--config-file` & `--node-config-file` unset
    * Current behavior for both **collector** and **report**.
  * `--config-file` set to`/etc/sos/custom1.conf`
    * **collector** uses `custom1.conf`
    * **report** uses `sos.conf`
  * `--node-config` set to `/etc/sos/custom2.conf`
    * **collector** uses default `/etc/sos/sos.conf`
    * **report** uses (pre-existing) `/etc/sos/custom2.conf` for the local and remote nodes .
* `--inherit-config-file` is `True`
  * `--config-file` & `--node-config-file` unset
    * **collector** uses `/etc/sos/sos.conf`
    * `/etc/sos/sos.conf` is copied to all nodes in a temp path.
    * **report** uses `/var/tmp/sos.XXXXX/sos.conf` (local could use default path)
  * `--config-file` set to`/etc/sos/custom3.conf`
    * **collector** uses `/etc/sos/custom3.conf`
    * `/etc/sos/custom3.conf` is copied to all nodes in a temp path.
    * **report** uses `/var/tmp/sos.XXXXX/custom3.conf` (local ...)

`--node-config-file` & `--inherit-config-file are an incompatible argument pair. Potentially also `--config-file` & `--node-config-file`.

Does this correctly describe what you were thinking?

Reply:
I've got this logic and additional comments worked out. I will add the new commit for review, but can be dropped/reverted if this wasn't what you were thinking of.